plugins/modules/credential.py

```diff
@@ -145,13 +158,18 @@ def main() -> None:
 
     argument_spec.update(AUTH_ARGSPEC)
 
-    required_if = [
-        (
-            "state",
-            "present",
-            ("name", "credential_type_name", "inputs", "organization_name"),
-        )
-    ]
+    module = AnsibleModule(argument_spec=argument_spec)
```

Comment:
Can you please also pass `supports_check_mode=True` to the constructor?

Reply:
added.